bin/codemods/src/assign-component-to-displayname-variable.js

```diff
@@ -0,0 +1,157 @@
+/*
+ This codemod updates
+
+ export default someHoC( React.createClass( {
+    displayName: 'SomeDisplayName',
+ } ) );
+
+ to
+
+ const SomeDisplayName = React.createClass({
+   displayName: 'SomeDisplayName',
+ });
+
+ export default someHoC( SomeDisplayName );
+
+ and
+
+ export default someHoC( class extends {Component|PureComponent|React.Component} {
+    static displayName = 'SomeDisplayName';
+ } );
+
+ to
+
+ class SomeDisplayName extends {Component|PureComponent|React.Component} {
+    static displayName = 'SomeDisplayName';
+ };
+
+ export default someHoC( SomeDisplayName );
+ */
+
+const createClassIdentifier = {
+    type: 'CallExpression',
+    callee: {
+        type: 'MemberExpression',
+        object: {
+            name: 'React',
+        },
+        property: {
+            name: 'createClass',
+        }
+    },
+};
+
+const displayNamePropertyIdentifier = {
+    key: {
+        name: 'displayName',
+    },
+};
+
+const isValidComponentString = string =>
+    string === 'Component' ||
+    string === 'PureComponent';
+
+const argIsCreateClassInstance = arg =>
+        arg.get( 'callee' , 'object', 'name' ).value === 'React' &&
+        arg.get( 'callee' , 'property', 'name' ).value === 'createClass';
+
+const argIsExtendsComponentInstance = arg =>
+    arg.get( 'type' ).value === 'ClassExpression' && (
+        isValidComponentString( arg.get( 'superClass', 'name' ).value ) ||
+        arg.get( 'superClass', 'object', 'name' ).value === 'React' &&
+        isValidComponentString( arg.get( 'superClass', 'property', 'name' ).value )
+    );
+
+const argIsComponent = arg =>
+    argIsCreateClassInstance( arg ) ||
+    argIsExtendsComponentInstance( arg );
+
+const getMatchingArg = args =>
+    args.filter( argIsComponent )[ 0 ];
+
+export default function transformer( file, api ) {
+    const j = api.jscodeshift;
+    const root = j( file.source );
+
+    // string => object =>
+    const replaceClassOrGetValue = displayName => arg =>
+        argIsComponent( arg )
+            ? j.identifier( displayName )
+            : arg.value;
+
+    const exportDefaultDeclarations = root
+        .find( j.ExportDefaultDeclaration );
+
+    exportDefaultDeclarations.forEach( node => {
+        const calleeNameValue = node.get( 'declaration', 'callee', 'name' ).value
+
+        if ( ! calleeNameValue ) {
+            throw new Error( 'return early, no immediate function call' );
```

Comment:
Throwing errors looks pretty awful so I tried `console.log` which looks pretty nasty too - is it worth having anything here at all or better just to allow this to fail silently?

Reply:
Can you give me an example of what `calleeNameValue` represents?  It doesn't seem to be like the lack of it should be an error.  Usually if a codemod doesn't need to modify a file then it just returns the original `file.source` unmodified

Reply:
`calleeNameValue` Would be `localize( ... )` or any other wrapping function.
I'd seen errors been thrown in (IIRC) some react codemod. I think it makes most sense to just return early without making any noise, then as you say, return the original / don't call `.replaceWith`

Reply:
IMO, error should mean that there is something _really_ wrong.  like an issue with the source file, or an unexpected edge case (multiple default exports).  Standard noop flow can be silent

Reply:
Yeah I'm feeling that too, I'll squish out those errors/logs and just have it return early :)